src/config/ConfigurationProvider.js

```diff
@@ -0,0 +1,109 @@
+/** @ignore @typedef {import("./Configuration.js").Configuration} Configuration */
+
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import log4js from 'log4js';
+import JSON5 from 'json5';
+
+/**
+ * Loads a linter {@link Configuration} from file
+ * @interface
+ */
+export class ConfigurationProvider {
+    /**
+     * @interface
+     * @returns {Promise<Configuration>}
+    */
+    async loadConfiguration() { throw new Error("not implemented"); }
+}
+
+/** @implements {ConfigurationProvider} */
+export class FileConfigurationProvider {
+    static logger = log4js.getLogger("i18nlint.FileConfigurationProvider");
+
+    /** @private @readonly */ filePath;
+
+    /**
+     * @param {string} filePath file to load configuration from
+     */
+    constructor(filePath) {
+        this.filePath = filePath;
+    }
+
+    /** @protected */
+    async loadJsonConfiguration() {
+        const content = await fs.readFile(this.filePath, "utf-8");
```

Comment:
As discussed: `ConfigurationProvider.loadJsConfiguration` uses an ESM dynamic import because it's basically the only way to load a JS file config. This means that `.loadConfiguration` has to be async (and this propagates all the way up the call stack) so we might as well use async `fs` calls at this point.